sycl/source/detail/context_impl.cpp

```diff
@@ -206,6 +208,26 @@ pi_native_handle context_impl::getNative() const {
   return Handle;
 }
 
+bool context_impl::isBufferLocationSupported() const {
+  if (SupportBufferLocationByDevices != NotChecked)
+    return SupportBufferLocationByDevices == Supported ? true : false;
+  // Check that devices within context have support of buffer location
+  size_t return_size = 0;
+  pi_device_info device_info;
+  SupportBufferLocationByDevices = Supported;
+  auto Plugin = getPlugin();
+  for (auto &Device : MDevices) {
+    const RT::PiDevice PiDevice = getSyclObjImpl(Device)->getHandleRef();
+    if (Plugin.call_nocheck<detail::PiApiKind::piDeviceGetInfo>(
+            PiDevice, (pi_device_info)PI_MEM_PROPERTIES_ALLOC_BUFFER_LOCATION,
+            sizeof(pi_device_info), &device_info, &return_size) != PI_SUCCESS) {
+      SupportBufferLocationByDevices = NotSupported;
+      break;
+    }
+  }
```

Comment:
I think this specific way of calling get device info will always return false since the lower level runtime does not support passing `PI_MEM_PROPERTIES_ALLOC_BUFFER_LOCATION` into the get device info API. Is there no other way of determining whether the target is an FPGA?

If we do this, then the OpenCL Spec has to be changed.

Reply:
Although technically the opencl spec is written in a way that enforces all vendor's runtime to accept this property, so technically it should be fine to directly pass it into other runtime (although they may do nothing with the property).

Reply:
`Is there no other way of determining whether the target is an FPGA that does not rely on lower level runtime?`
We can determine if the device is FPGA or not in runtime. I tried to do this in https://github.com/intel/llvm/pull/5604/commits/ccf73a63822962464eec97433e86daf8d4cbacbb
Do you think it would be better to just check if a device is FPGA or not?

Reply:
Yes I do think it is better to check if a device is FPGA directly, because even if it error out FPGA device, it will prompt the other vendor to support passing in of buffer location feature in their runtime API.